osquery/filesystem/darwin/bsd_file_flags.cpp

```diff
@@ -0,0 +1,77 @@
+/**
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed in accordance with the terms specified in
+ *  the LICENSE file found in the root directory of this source tree.
+ */
+
+#include <iomanip>
+#include <unordered_map>
+
+#include <boost/algorithm/string/join.hpp>
+
+#include <sys/stat.h>
+
+#include <osquery/filesystem/fileops.h>
+
+namespace osquery {
+namespace {
+/// The list of supported flags, as documented in `man 2 chflags`
+const std::unordered_map<std::uint32_t, std::string> kBsdFlagMap = {
+    {UF_NODUMP, "NODUMP"},
+    {UF_IMMUTABLE, "UF_IMMUTABLE"},
+    {UF_APPEND, "UF_APPEND"},
+    {UF_OPAQUE, "OPAQUE"},
+    {UF_HIDDEN, "HIDDEN"},
+    {SF_ARCHIVED, "ARCHIVED"},
+    {SF_IMMUTABLE, "SF_IMMUTABLE"},
+    {SF_APPEND, "SF_APPEND"}};
+
+std::uint32_t getBsdFlagMask() {
+  std::uint32_t result = 0U;
+
+  for (const auto& p : kBsdFlagMap) {
+    const auto& bit = p.first;
+    result |= bit;
+  }
+
+  return result;
+}
+} // namespace
+
+Status describeBSDFileFlags(std::string& output, std::uint32_t st_flags) {
+  output.clear();
+
+  static const auto flag_mask = getBsdFlagMask();
+
+  std::vector<std::string> label_list;
+
+  for (const auto& p : kBsdFlagMap) {
```

Comment:
Will this always iterate in the same order? I think it's desirable for `output` to have a stable order.